scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/env python3.8
+
+import argparse
+import json
+import os
+import subprocess
+
+from typing import List, Union, Any
+from deepdiff import DeepDiff
+
+# {{{ JsonRpcProcess
+class MyEncoder(json.JSONEncoder):
+    """
+    Encodes an object in JSON
+    """
+    def default(self, o): # pylint: disable=E0202
+        return o.__dict__
+
+class JsonRpcProcess:
+    exe_path: str
+    exe_args: List[str]
+    process: subprocess.Popen
+    trace_io: bool
+
+    def __init__(self, exe_path: str, exe_args: List[str], trace_io: bool = False):
+        self.exe_path = exe_path
+        self.exe_args = exe_args
+        self.trace_io = trace_io
+
+    def __enter__(self):
+        self.process = subprocess.Popen(
+            [self.exe_path, *self.exe_args],
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        return self
+
+    def __exit__(self, exception_type, exception_value, traceback) -> None:
+        self.process.kill()
+        self.process.wait(timeout=2.0)
+
+    def trace(self, topic: str, message: str) -> None:
+        if self.trace_io:
+            print(f"{SGR_TRACE}{topic}:{SGR_RESET} {message}")
+
+    def receive_message(self) -> Any:
+        # `, timeout: float = 2.0`
+        # Note, we should make use of timeout to avoid infinite blocking if nothing is received.
+        LEN_HEADER = "Content-Length: "
+        TYPE_HEADER = "Content-Type: "
```

Comment:
Why not use full names? Had to scroll back here to check what these headers meant and then realized that it would have been obvious if you just used full names.
```suggestion
        CONTENT_LENGTH_HEADER = "Content-Length: "
        CONTENT_TYPE_HEADER = "Content-Type: "
```